tests/test_gcm_push_payload.py

```diff
@@ -1,76 +1,76 @@
-				"application/json", application_id=None)
-
-	def test_fcm_push_payload_params(self):
-		with mock.patch("push_notifications.gcm._fcm_send", return_value=GCM_JSON) as p:
-			send_message(
-				"abc",
-				{"message": "Hello world", "title": "Push notification", "other": "misc"},
-				"FCM",
-				delay_while_idle=True, time_to_live=3600, foo="bar",
-			)
-			p.assert_called_once_with(
-				b'{"data":{"other":"misc"},"delay_while_idle":true,'
-				b'"notification":{"body":"Hello world","title":"Push notification"},'
-				b'"registration_ids":["abc"],"time_to_live":3600}',
-				"application/json", application_id=None)
+			send_message("abc", message)
+
+			# one call
+			assert len(p.mock_calls) == 1
+			call = p.mock_calls[0]
+
+			# only messages is args, dry_run and app are in kwargs
+			assert len(call.args) == 1
+
+			assert "dry_run" in call.kwargs and call.kwargs["dry_run"] is False
+			assert "app" in call.kwargs and call.kwargs["app"] is None
+
+			# only one message
+			assert len(call.args[0]) == 1
+
+			message = call.args[0][0]
+			assert isinstance(message, Message)
+			assert message.token == "abc"
+			assert message.android.notification.body == "Hello world"
 
 	def test_fcm_push_payload_many(self):
-		with mock.patch("push_notifications.gcm._fcm_send", return_value=GCM_JSON_MULTIPLE) as p:
-			send_bulk_message(["abc", "123"], {"message": "Hello world"}, "FCM")
-			p.assert_called_once_with(
-				b'{"notification":{"body":"Hello world"},"registration_ids":["abc","123"]}',
-				"application/json", application_id=None)
-
-	def test_gcm_push_payload(self):
-		with mock.patch("push_notifications.gcm._gcm_send", return_value=GCM_JSON) as p:
-			send_message("abc", {"message": "Hello world"}, "GCM")
-			p.assert_called_once_with(
-				b'{"data":{"message":"Hello world"},"registration_ids":["abc"]}',
-				"application/json", application_id=None)
-
-	def test_gcm_push_payload_params(self):
-		with mock.patch("push_notifications.gcm._gcm_send", return_value=GCM_JSON) as p:
-			send_message(
-				"abc", {"message": "Hello world"}, "GCM",
-				delay_while_idle=True, time_to_live=3600, foo="bar",
-			)
-			p.assert_called_once_with(
-				b'{"data":{"message":"Hello world"},"delay_while_idle":true,'
-				b'"registration_ids":["abc"],"time_to_live":3600}',
-				"application/json", application_id=None)
-
-	def test_gcm_push_payload_many(self):
-		with mock.patch("push_notifications.gcm._gcm_send", return_value=GCM_JSON_MULTIPLE) as p:
-			send_bulk_message(["abc", "123"], {"message": "Hello world"}, "GCM")
-			p.assert_called_once_with(
-				b'{"data":{"message":"Hello world"},"registration_ids":["abc","123"]}',
-				"application/json",
-				application_id=None)
+		with mock.patch("firebase_admin.messaging.send_all", return_value=FCM_SUCCESS) as p:
+			message = dict_to_message({"message": "Hello world"})
+
+			send_message(["abc", "123"], message)
+
+			# one call
+			assert len(p.mock_calls) == 1
```

Comment:
when not using the assert methods from a mock, `self.assertEqual()` or similar should be preferred (mostly for better error reporting)